astarte-device-sdk/AstarteDeviceSDK.h

```diff
@@ -32,6 +32,8 @@ class AstarteGenericConsumer;
 class AstarteGenericProducer;
 class QJsonSchemaChecker;
 
+enum endpointType {Scalar, Array};
```

Comment:
`Scalar` and `Array` names are too generic, they might conflict with other stuff, lets use `AstarteScalarType` or stuff like that.